js/src/dapps/registry/events/actions.js

```diff
@@ -0,0 +1,31 @@
+export const start = (name, from, to) => ({ type: 'events subscribe start', name, from, to });
+
+export const event = (name, event) => ({ type: 'events event', event: { ...event, type: name } });
+
+export const fail = (name) => ({ type: 'events subscribe fail', name });
+
+export const subscribe = (name, from = 0, to = 'latest') =>
+  (dispatch, getState) => {
+    const { contract } = getState();
+    if (!contract || !contract.instance) return;
+
+    dispatch(start(name, from, to));
+    contract.subscribe(name, { fromBlock: from, toBlock: to }, (err, events) => {
+      if (err) {
+        console.error(`could not subscribe to event ${name}.`);
```

Comment:
Hm. Right now the registry dapp is pretty small. And using stuff from the main code will make it difficult to separate the dapps.